raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -25,55 +25,60 @@ contract TokenNetwork is Utils {
     // Channel identifier is a uint256, incremented after each new channel
     mapping (uint256 => Channel) public channels;
 
+    // We keep the balance data in a separate mapping to allow channel data structures to be
+    // removed when settling the channel. If there are locked transfers, we need to store
+    // balance data to be able to unlock them.
+    // Key is a hash of the channel_identifier and participant address.
+    mapping(bytes32 => BalanceData) balance_data;
+
     // Used for determining the next channel identifier
     // Start from 1 instead of 0, otherwise the first channel will have an additional
     // 15000 gas cost than the rest
     uint256 public last_channel_index = 0;
 
-    // The ClosingRequest identfier must be THE SAME as the channel identifier (mapping key) from `channels`
-    mapping (uint256 => ClosingRequest) public closing_requests;
-
-    struct Participant
-    {
+    struct Participant {
         // Total amount of token transferred to this smart contract through the
         // `setDeposit` function, note that direct token transfer cannot be
         // tracked and will be burned.
-        uint256 deposit;
-
-        // The latest known merkle root of the pending hash-time locks, used to
-        // validate the withdrawn proofs.
-        bytes32 locksroot;
+        uint240 deposit;
 
-        // The latest known transferred_amount from this node to the other
-        // participant, used to compute the net balance on settlement.
-        uint256 transferred_amount;
-
-        // Value used to order transfers and only accept the latest on calls to
-        // update, this will only be relevant after either #182 or #293 is
-        // implemented.
-        uint64 nonce;
-
-        // This value is set to True after the channel has been opened.
+        // This value is set to true after the channel has been opened.
         // This is an efficient way to mark the channel participants without doing a deposit.
-        // This is uint8 and it gets packed with the nonce.
+        // This is bytes1 and it gets packed with the rest of the struct data.
         bool initialized;
 
-        // A mapping to keep track of locks that have been unlocked.
-        mapping(bytes32 => bool) unlocked_locks;
+        // This is a value set to true after the channel has been closed, only if this is the
+        // participant who closed the channel.
+        // This is bytes1 and it gets packed with the rest of the struct data.
+        bool is_closer;
     }
 
-    struct Channel {
-        // We also use the settle_timeout to verify that
-        // the channel is open
-        uint256 settle_timeout;
-        mapping(address => Participant) participants;
+    struct BalanceData {
```

Comment:
> why not two struct definitions?

In terms of gas cost, you will end up with 10k more gas when calling `settleChannel`
now: 5k * 2 (2 SSTORE for modifying the 2 variables in `BalanceData`) = 10k
2 structs: - 10k * 2 (deleting 1 struct with `balance_hash`, `nonce`) + 20k * 2 (adding the 2nd struct) = 20k

But if the logic is not clear enough, it can be considered.

Reply:
> And what about renaming it to BalanceProofData?

It is `BalanceProofData` for `balance_hash` and `nonce` and `BalanceData` for `locksroot` and `locked_amount`. At least this is how we agreed in https://github.com/raiden-network/spec/pull/54 .
If we combine them in 1 struct, as it is now `BalanceData` seems more general to me + it's shorter. But if we get another opinion saying otherwise, I'll change.